### DISCUSSION TITLE
Documentation about Dialog component adds confusion regarding modal vs non-modal

### DISCUSSION THREAD

**Discussion by linasburneika** [iTwin/iTwinUI#1686] [Feedback]
Dialog component is described as "purposefully interruptive", the [first Dialog example ](https://itwinui.bentley.com/docs/dialog) demonstrates a modal dialog. To make it non-modal it's expected to remove `<Dialog.Backdrop />` element from the composition. However section [Dialog vs Modal](https://itwinui.bentley.com/docs/dialog#dialog-vs-modal) shows a code snippet with older `<Modal/>` component, which is not documented separately anymore. I think documentation should state that the `<Dialog/>` is a generic component, which can be configured to work as modal dialog or non-blocking regular dialog, and don't show usage of `<Modal/>` at all. For example documentation of native `<dialog>` tag makes it very clear: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog

itwinui.bentley.com/docs/dialog doesn't have a direct example of non-modal dialog without a backdrop, there is such a case in the storybook.

Modal vs non-modal is confusing especially for those migrating from older versions of iTwinUI. If you used `<Modal/>` before, are you supposed to change it to the Dialog? If you started a new project, is it preferable to use Dialog component in all the cases?

Documentation about `<Modal/>` component is missing in itwinui.bentley.com. If this is a deliberate choice to promote usage of Dialog, the migration guide should have a note about it.

**Comment by mayank99** [comment]
This is good feedback! The content on that page was hastily copied from the old UX site and was written before we had the `Dialog` component. We'll rewrite the page to make everything clearer.

And `Modal` is not deprecated, so it's totally fine to keep using it. In addition to adding a backdrop, it also preconfigures other things like trapping focus, closing on outside click, disabling page scroll, etc. We'll document this too.
